### DISCUSSION TITLE
How to create one input and two input ?

### DISCUSSION THREAD

**Discussion by bobykurniawan11** [xyflow/xyflow#4337] [Q&A] [answered]
Here is what i do

```
import { useCallback } from 'react';
import { Handle, NodeResizer, Position } from 'reactflow';

const handleStyle = { left: 10 };

export function CustomNodeQuestion({ data }: any) {
    return (
        <div className='w-full h-full px-4 py-2 shadow-md rounded-md bg-white border-2 border-stone-400'>
            <NodeResizer minWidth={100} minHeight={30} />
            <Handle type="target" position={Position.Left} />
            <div className='flex flex-col mt-4 gap-4'>
                <div className='flex flex-col mt-4'>
                    <p className='text-xs text-a'>Option 1</p>
                    <Handle id='source-option-1' type="source" position={Position.Right} />
                </div>
                <div className='flex flex-col mt-4'>
                    <p className='text-xs text-a'>Option 2</p>
                    <Handle id='source-option-2' type="source" position={Position.Right} />
                </div>
            </div>
        </div>
    );
}
```

here is the result

![Screenshot 2024-06-05 at 14 21 15](https://github.com/xyflow/xyflow/assets/21689649/6c4276b5-5d50-4715-81f2-78bddd084942)

My expecting result is like this

![Screenshot 2024-06-05 at 14 40 47](https://github.com/xyflow/xyflow/assets/21689649/a73c45bc-107f-47d7-ab9c-48218fde777d)

**Comment by bcakmakoglu** [comment]
[You should check the `Custom Node` example in detail.](https://reactflow.dev/examples/nodes/custom-node)

TL;DR: 
Use css to adjust the exact position of a `<Handle>`
```tsx
<Handle
  type="source"
  position={Position.Right}
  id="a"
  style={{ top: 10, background: '#555' }}
  isConnectable={isConnectable}
/>
<Handle
  type="source"
  position={Position.Right}
  id="b"
  style={{ bottom: 10, top: 'auto', background: '#555' }}
  isConnectable={isConnectable}
/>

**Reply by bobykurniawan11** [reply]
HI, thanks for replying.

I wonder, when i add connect to another node. When i choose source from id `b` to dragging the line. The source still from id `a`

**Reply by bcakmakoglu** [reply]
Not sure I understand what you're saying? You're dragging from handle b and it creates a connection from handle a? 🤔
Seems to work completely fine for me:

https://github.com/xyflow/xyflow/assets/78412429/747753d1-1317-4a8f-96bb-35757abcc4ae

**Reply by bobykurniawan11** [reply]
Sorry for my english here is what i mean


https://github.com/xyflow/xyflow/assets/21689649/c12ad5bd-2080-4244-8328-43a331313537

I think it is because my other custom node
```
export function CustomEndNode({ data }: any) {
    const onChange = useCallback((evt: any) => {
        console.log(evt.target.value);
    }, []);

    return (
        <div className='w-full h-full py-2 px-8'>
            <Handle id='dead-end' type="target" position={Position.Left} isConnectable={true} isConnectableEnd={true} />

            <div className='flex flex-col mt-4'>
                <p className='text-xs text-a'>End Node</p>
                <p className='text-xs text-a'>
                    {
                        data?.label
                    }
                </p>
            </div>
        </div>
    );
}
```

**Reply by bcakmakoglu** [reply]
Most likely you're doing something wrong in your `onConnect` handler. What does it look like?

**Reply by bcakmakoglu** [reply]
Also do your handles have a unique id? In the code snippet above they have an id but who knows, maybe your actual code doesn't have an id on the handles, which they *require* if you use more than one `<Handle>` of the same type (`source` or `target`)

**Reply by bobykurniawan11** [reply]
Here is my onConnect

```

    const onConnect = useCallback(
        (params: Connection) => {
            const anotherUniqueID = getId();
            setEdges((eds: any) => eds && eds.concat({
                id: `e${params.source}-${id}`, source: params.source, target: params.target, data: {
                    // label: `Jawaban Di sini`,
                },
                labelBgStyle: { fill: 'white' },
                labelStyle: { fill: 'red', fontWeight: 700, fontSize: 12 },
            }));

        },
        [setEdges]
    );

```

My on drop

```
    const onDrop = useCallback(
        (event: any) => {
            console.log("ON ROP!!!")
            event.preventDefault();
            const type = event.dataTransfer.getData('application/reactflow');
            if (typeof type === 'undefined' || !type) {
                return;
            }
            const position = rfInstance.screenToFlowPosition({
                x: event.clientX,
                y: event.clientY,
            });
            const newNode = {
                id: getId(),
                type,
                position,
                data: { label: `${type} node` },
            };
            setNodes((nds) => nds.concat(newNode));
        },
        [rfInstance],
    );
```

**Reply by bcakmakoglu** [reply, marked as answer]
As suspected, you're not passing the `sourceHandle` and `targetHandle` properties to the edge you create, how is the edge supposed to know what `<Handle>` to use if you never pass that information on? 😄 

Change this code
```ts
const onConnect = useCallback(
    (params: Connection) => {
        const anotherUniqueID = getId();
        setEdges((eds: any) => eds && eds.concat({
            id: `e${params.source}-${id}`, source: params.source, target: params.target, data: {
                // label: `Jawaban Di sini`,
            },
            labelBgStyle: { fill: 'white' },
            labelStyle: { fill: 'red', fontWeight: 700, fontSize: 12 },
        }));

    },
    [setEdges]
);
```

To this:
```ts
const onConnect = useCallback(
  (params: Connection) => {
    const id = getId();

    setEdges((eds) => {
      return [
        ...eds,
        {
          id: `e${params.source}-${id}`,
          ...params, // <--- Pass *all* params to the new edge, not just `source` and `target`!
          data: {
            // label: `Jawaban Di sini`,
          },
          labelBgStyle: { fill: 'white' },
          labelStyle: { fill: 'red', fontWeight: 700, fontSize: 12 },
        },
      ];
    });
  },
  [setEdges]
);
```

**Reply by bobykurniawan11** [reply]
Thank you so much! You're saving me a day's worth of debugging."
